cachi2/core/package_managers/pip.py

```diff
@@ -1704,6 +1772,7 @@ def _download_vcs_package(requirement: PipRequirement, pip_deps_dir: RootedPath)
     return {
         "package": requirement.package,
         "path": download_to.path,
+        "subpath": requirement.url.split('#')[-1],
```

Comment:
The subpath looks a bit different https://pip.pypa.io/en/stable/topics/vcs-support/#url-fragments

```
pip install "pkg @ vcs+protocol://repo_url/#subdirectory=pkg_dir"
```

But now that I look at it, the pip style of vcs urls conflicts with [PEP 440](https://peps.python.org/pep-0440/#direct-references), where the fragment means commit hash

In the docs, we claim that we support "a subset of PEP 440" https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#external-dependencies

So I wouldn't mind if we don't support subpath (TBH, I'm pretty sure that our way of supporting external dependencies is incompatible with subpaths anyway)